docs/user-interface/handlers/customize.md

```diff
@@ -1,122 +1,316 @@
 
-Handlers are global and therefore customization can occur anywhere in your .NET MAUI app. For example, customizing a handler for a control in your `App` class will result in all controls of that type being customized in the app. Similarly, customizing a handler for a control in the first page of your app will also result in all controls of that type in the app being customized.
+Handlers can be customized to augment the appearance and behavior of a .NET MAUI control beyond the customization that's possible through the control's API. Handlers are global and therefore customization can occur anywhere in your .NET MAUI app. For example, customizing a handler for a control in your `App` class will result in all controls of that type being customized in the app. Similarly, customizing a handler for a control in the first page of your app will also result in all controls of that type in the app being customized.
 
-## Examples
+Handlers can be customized per platform by using compiler preprocessor directives, to multi-target code based on the platform. Alternatively, you can use partial classes to organize your code into platform-specific folders and files. For more information about conditional compilation, see [Conditional compilation](/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation).
 
-Handlers are typically customized to augment the appearance and behavior of a native platform control. Handlers can be customized per platform by using the `#ifdef` compiler directive, to multi-target code based on the platform. However, you can just as easily use platform-specific folders or files to organize such code.
+## Handler lifecycle
 
-> [!NOTE]
-> Handler customization can range from the simple to the complex. More complex examples will appear in time.
+All handler-based .NET MAUI controls support two handler lifecycle events:
 
-### Customize the background color of all controls
+- `HandlerChanged` is fired after the handler for a cross-platform control has been created. This event indicates that the native control that implements the cross-platform control is available, and all the property values set on the cross-platform control have been applied to the native control.
+- `HandlerChanging` is fired before an existing handler is removed from a cross-platform control, and before a new handler for the cross-platform control is created. This event indicates that the existing native control is about be removed from the cross-platform control, and therefore that any native events should be unwired and other cleanup performed. The `HandlerChangingEventArgs` object that accompanies this event has `OldHandler` and `NewHandler` properties, of type `IElementHandler`.
```

Comment:
I would be clear on the `HandlerChanging` and `HandlerChanged` events. to me it sounds like this is true and may be worth calling out (especially if I'm wrong in these assumptions):

01. The `HandlerChanging` event is raised before the `HandlerChanged` event.
01. This doesn't happen the first time handlers are added to controls, since there is nothing being "changed" (removed before an add).

Reply:
(2) is wrong so I'll clarify.